src/types/app_context.go

```diff
@@ -13,20 +10,13 @@ type AppContext struct {
 	Config   AppConfig
 }
 
-// GetServiceContext returns a ServiceContext for the service found at the passed location
-func (a AppContext) GetServiceContext(location string) (ServiceContext, error) {
-	serviceConfig := ServiceConfig{}
-	yamlFile, err := ioutil.ReadFile(path.Join(location, "service.yml"))
-	if err != nil {
-		return ServiceContext{}, err
-	}
-	if err = yaml.Unmarshal(yamlFile, &serviceConfig); err != nil {
-		return ServiceContext{}, err
-	}
+// GetServiceContext returns a ServiceContext for the service found at the given directory base
+func (a AppContext) GetServiceContext(serviceDirBase string) (ServiceContext, error) {
+	serviceConfig, err := NewServiceConfig(a.Location, serviceDirBase)
```

Comment:
Why pass in the appDir and serviceDirName separately? Can we join them once here (since that is also happening a few lines down) and pass in the full serviceDir into `NewServiceConfig`